File: api/internal/dto/response/CourseResponse.go
package response

type CourseResponse struct {
	Id          uint                      `json:"id"`
	Code        string                    `json:"code"`
	Name        string                    `json:"name"`
	Description string                    `json:"description"`
	Avatar      string                    `json:"avatar"`
	Background  string                    `json:"background"`
	Visibility  string                    `json:"visibility"`
	Teacher     TeacherPublicInfoResponse `json:"teacher"`
}

File: api-v2/smart-elearning/src/main/java/com/smartelearning/smartelearning/dto/response/CourseResponse.java
/*
 *  CourseResponse
 *  @author: minhhieuano
 *  @created 7/10/2025 4:07 PM
 * */


package com.smartelearning.smartelearning.dto.response;

import com.smartelearning.smartelearning.common.enums.CoursePermission;
import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class CourseResponse {
    private String id;
    private String avatar;
    private String name;
    private String description;
    private String background;
    private CoursePermission permission;
    private UserPublicInfoResponse teacher;
    private Integer lessonCount;
    private Integer studentCount;
}
